### DISCUSSION TITLE
Firebase Admin SDK for DART

### DISCUSSION THREAD

**Discussion by alexsandercdm** [firebase/flutterfire#9526] [Feature request]
Why is there no Firebase admin SDK for dart?

I believe it is necessary for development and integration with my backend in dart, are there any plans to make this tool available?

**Comment by kylehayes** [comment]
Yes, exactly. It is quite surprising to me that a Google created language does not support a Google created backend natively without dependence on flutter.

**Comment by marb2000** [comment]
Thank you for opening this feature request. To help us prioritize this request over others we need more information about the scenario and what are the pain points of not having Dart support in the Admin SDK. 

For example, I need to use the Firebase products X, Y, and Z, and not supporting Dart make us to slow down our product development given we don't have expertise on JS, Java, etc. Also, it causes issues with ....

We are listening every single request and we want to understand bettter your scenario.

**Reply by kylehayes** [reply]
I'm happy to provide some detail here from our perspective. We set off to create an application in Dart with Flutter. Once we assessed the need to create a backend, we had hoped to keep the whole codebase in a single language since our development team as an early stage startup was so small (two engineers in total). We had begun to do this for about a month with a variety of other functions that weren't database centric.

We then realized the Firebase logic we had on the frontend was not going to be compatible with the backend which caused us to investigate all of our options writing Dart on the server side. Ultimately we backed out of the decision to keep all as a single language due to the immaturity of server-side Dart and opted instead for a Go built API backed by Cloud SQL.

**Reply by marb2000** [reply]
Thanks for the details! @kylehayes

**Reply by ayazemregm** [reply]
It is insane that team did not see value in this. We use flutter and we use Dart on back end as well. We end up creating our wrappers around the API. We would like to manage users, their custom fields and all sorts of administrative functions on our back end. Do you have anything planned for this?

**Reply by ayazemregm** [reply]
Also, REST API does not have all of the functionality. We ended up forking current community package and edit it for our use case.

**Comment by pierremrtn** [comment]
I need to use firebase admin sdk to update various user fields from the server when some api call are made. I would like to write the backend code in Dart since I'm already using it for front-end and don't have much experience with other languages. Not supporting Dart for admin task is quite annoying.

**Reply by diezep** [reply]
+1

**Comment by mafreud** [comment]
While this is WIP, invertase is developing [dart_firebase_admin](https://pub.dev/packages/dart_firebase_admin).

**Reply by cedvdb** [reply]
why not just access a real sdk with ffi ? I don't see the advantage of doing everything from scratch in dart

**Reply by mafreud** [reply]
a real sdk?

**Reply by cedvdb** [reply]
the go-firebase-admin-sdk or the node-firebase-admin-sdk.  EG an effort to do something like this https://pub.dev/packages/firebase_go_admin
